mesh/mesh.go

```diff
@@ -432,6 +433,13 @@ func (msh *Mesh) pushLayersToState(ctx context.Context, from, to types.LayerID)
 	missing := msh.MissingLayer()
 	// we never reapply the state of oldVerified. note that state reversions must be handled separately.
 	for layerID := from; !layerID.After(to); layerID = layerID.Add(1) {
+		if !layerID.After(msh.latestLayerInState) {
+			logger.With().Warning("trying to apply layer before currently applied layer",
+				log.Stringer("applied_layer", msh.latestLayerInState),
+				layerID,
+			)
+			continue
```

Comment:
why not immediately return an error? do you mean to skip layers that are <= applied and then apply those higher? this will result in state root difference. right?

Reply:
> do you mean to skip layers that are <= applied and then apply those higher? 

yes

> this will result in state root difference. right?

it may, but not necessarily.
i think it is not as bad as it seems, because after next rerun (+1 more hour) state diff will be resolved anyway